api/internal/localizer/localizer.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package localizer
+
+import (
+	"log"
+	"path/filepath"
+
+	"sigs.k8s.io/kustomize/api/ifc"
+	plgnsLdr "sigs.k8s.io/kustomize/api/internal/plugins/loader"
+	"sigs.k8s.io/kustomize/api/internal/target"
+	"sigs.k8s.io/kustomize/api/konfig"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+	"sigs.k8s.io/yaml"
+)
+
+// Localizer encapsulates all state needed to localize the root at ldr.
+type Localizer struct {
+	fSys filesys.FileSystem
+
+	// kusttarget fields
+	validator ifc.Validator
+	rFactory  *resmap.Factory
+	pLdr      *plgnsLdr.Loader
+
+	// underlying type is LocLoader
+	ldr ifc.Loader
```

Comment:
This field may need to store the output of `LocLoader.New`, which is an `ifc.Loader`. `LocLoader.New` cannot output `LocLoader` instead because we want `LocLoader` to implement `ifc.Loader`.

The only way around storing `ifc.Loader` instead of `LocLoader` that I can think of is a type assertion, but I find that ugly? Would love to hear suggestions.